src/module.c

```diff
@@ -6633,21 +6630,17 @@ int RM_BlockedClientDisconnected(RedisModuleCtx *ctx) {
  * the module ID and thus be more useful for logging. */
 RedisModuleCtx *RM_GetThreadSafeContext(RedisModuleBlockedClient *bc) {
     RedisModuleCtx *ctx = zmalloc(sizeof(*ctx));
-    RedisModuleCtx empty = REDISMODULE_CTX_INIT;
-    memcpy(ctx,&empty,sizeof(empty));
-    if (bc) {
-        ctx->blocked_client = bc;
-        ctx->module = bc->module;
-    }
-    ctx->flags |= (REDISMODULE_CTX_THREAD_SAFE |
-                   REDISMODULE_CTX_USES_TEMP_CLIENT);
+    RedisModule *module = bc ? bc->module : NULL;
+    int flags = REDISMODULE_CTX_THREAD_SAFE;
+    if (!bc) flags |= REDISMODULE_CTX_NEW_CLIENT;
+    moduleCreateContext(ctx, module, flags);
```

Comment:
maybe add a comment here, explaining why for blocked clients we use `bc->thread_safe_ctx_client` (refer to the code below), and for other thread safe clients, we think we can afford to take the slow path.

Reply:
added a comment